vpr/src/base/vpr_context.h

```diff
@@ -163,7 +163,7 @@ struct DeviceContext : public Context {
     /* A writeable view of routing resource graph to be the ONLY database 
      * for routing resource graph builder functions.
      */
-    RRGraphBuilder rr_graph_builder{&rr_nodes, &rr_node_metadata, &rr_edge_metadata};
+    RRGraphBuilder rr_graph_builder{&rr_nodes, rr_segments, rr_switch_inf, &rr_node_metadata, &rr_edge_metadata};
```

Comment:
Now, ``rr_segments`` and ``rr_switch_inf`` should be internal data. You do not need them in the argument list.

See the example of node_lookup(). This is what we want to achieve:

https://github.com/verilog-to-routing/vtr-verilog-to-routing/blob/372c028e3ab5b4aa9710f1bf8bb7c86d9c8206d2/vpr/src/device/rr_graph_builder.h#L236

Finally, when clean-up is done,  there should be no argument in the constructor. All the input arguments you see will be internal data.

Reply:
@tangxifan 
So we are going remove rr_segment and rr_switch_inf from argument list now ?

Reply:
Yes. Make them as an internal data just as we did for the ``node_lookup_``